### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by adamaltman**

### CODE REVIEW

**Comment by adamaltman** [comment]
This PR is bad.

However, you need to improve your usage of git and github.  Create a feature branch (related to the issues I made), and then make a PR, and link it to the issue.  Then get it reviewed.  You shouldn't close the issue, when it's not closed.  It's closed when it is merged.

**Comment by adamaltman** [comment]
You need to do a global git ignore on .vagrant and .idea because you have a lot of files that shouldn't be in here...

**Comment by garcijo** [comment]
Ok, I'll start doing that right away. As for this pull request, should I merge it or should I ignore it and make a new one once I'm done with one of the issues?

**Comment by adamaltman** [comment]
The proper thing to do is not to merge a bad PR... You should at a minimum remove the files that are here (.vagrant and .idea).

File: src/Web/Action/HomePageAction.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            $spotify = new SpotifyFeed($this->spotify, $this->db);
+            $song = $spotify->newSong($user);
+            $spotify->setSong($song);
+            return $this->renderer->render($response, 'home.phtml', $args);
+        } else {
+            $response = $response->withRedirect("/login");
+            return $response;
+        }
+    }
+}
```
Comment (adamaltman) [inline src/Web/Action/HomePageAction.php:45]: part of code standards is to have a blank line at the end of the file...  (you can make that a setting in phpstorm)

File: src/Web/Action/MusicPageAction.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    ) {
+        $this->renderer = $renderer;
+        $this->spotify = $spotify;
+        $this->db = $db;
+    }
+
+    public function __invoke(Request $request, Response $response, $args)
+    {
+        // Verify if user is authenticated. If false, redirect to login
+        if (isset($_SESSION['user'])) {
```
Comment (adamaltman) [inline src/Web/Action/MusicPageAction.php:34]: I think perhaps a perfect case for middleware -- is user authenticated?  If not, redirect to the login.

**Comment by adamaltman** [comment]
Nice work overall. :+1:

**Comment by adamaltman** [comment]
I believe you have to proactively remove them from the project (once on the gitignore, any further changes won't be committed, but the original files are still there).  I think I would do something like: `git rm <file>`.

**Comment by garcijo** [comment]
This PR fixes issues #3: add namespacing and autoloading, #5: config settings to yaml, and #6: make actions

**Pull request merged**